File: tests/bench.py
import time
import pandas as pd

start = time.time()

df = pd.DataFrame({
    "id": range(10_000_000),
    "value": [i * 1.5 for i in range(10_000_000)]
})

print(f"Created in {time.time() - start:.3f} seconds")

File: tests/bench.nim
import
    times,
    strformat,
    sequtils,
    razor

let start = cpuTime()
var df = newDataFrame()
df["id"] = newSeries((0..<10_000_000).toSeq)
df["value"] = newSeries(toSeq(0..<10_000_000).map(proc(x: int): float = x.float * 1.5))
echo &"Created in {cpuTime() - start:.3f} seconds"
